pkg/skaffold/runner/diagnose.go

```diff
@@ -58,6 +58,21 @@ func (r *SkaffoldRunner) DiagnoseArtifacts(out io.Writer) error {
 		fmt.Fprintln(out, " - Dependencies:", len(deps), "files")
 		fmt.Fprintf(out, " - Time to list dependencies: %v (2nd time: %v)\n", timeDeps1, timeDeps2)
 
+		timeSyncMap1, err := timeToConstructSyncMap(ctx, r.Builder, artifact)
```

Comment:
why do we do this 2 times?
Also, is there a way to add tests for the L63 to L65?

Reply:
I think it's run twice so that the OS can warm the inode cache. The second run should normally be more representative.

Since there was no test for diagnose at all, I didn't bother adding one. Since this is creating (random) timings, the best we can achieve here is a smoke test. Do you feel strongly about it?

Reply:
i think the runtime object inspection things could be unit tested to make sure we dont' break this in future. But i dont feel strongly regarding it. 

```
if _, isNotSupported := err.(build.ErrSyncMapNotSupported); !isNotSupported {
				return errors.Wrap(err, "construct artifact dependencies")
}
```

Reply:
@tejal29 Can you explain a little more how you would like that code to be refactored?

I opened #2157 for some smoke tests of the diagnose subcommand. I think this has advantages over unit testing here, because we would have to set up a lot of mock objects for unit tests.